### PR TITLE
[DOCS] Correct relevance calculation for distance_feature

### PR DESCRIPTION
**Pull request by aetter**
The relevance score calculation for `distance_feature` appears to use `weight`, not `boost`. In most (all?) cases, `weight` seems to be `boost^2`. Sample calls and portions of responses below.

1. Configure the index and documents as listed [here](https://www.elastic.co/guide/en/elasticsearch/reference/7.9/query-dsl-distance-feature-query.html#distance-feature-index-setup).

2. Search with the default boost:

```
GET items/_search
{
  "query": {
    "bool": {
      "should": {
        "distance_feature": {
          "field": "location",
          "pivot": "1000m",
          "origin": [-71.3, 41.15],
          "boost": 1
        }
      }
    }
  }
}
```

Top result (document 2) has a score of 1.0.

3. Increase boost to 3:

```
GET items/_search
{
  "query": {
    "bool": {
      "should": {
        "distance_feature": {
          "field": "location",
          "pivot": "1000m",
          "origin": [-71.3, 41.15],
          "boost": 3
        }
      }
    }
  }
}
```

Top result now has a score of 9, not 3, like you'd expect based on the documentation.

4. Explain the result:

```
GET items/_explain/2
{
  "query": {
    "bool": {
      "should": {
        "distance_feature": {
          "field": "location",
          "pivot": "1000m",
          "origin": [-71.3, 41.15],
          "boost": 3
        }
      }
    }
  }
}

"value": 9.0,
"description": "Distance score, computed as weight * pivotDistance / (pivotDistance + abs(distance)) from:",
"details": [
  {
    "value": 9.0,
    "description": "weight",
    "details": []
  },
  ...
]
```

The explain description references `weight`, so it seems OK to use it in the documentation, as well.

5. The same logic appears to apply to dates:

```
GET items/_search
{
  "query": {
    "bool": {
      "should": {
        "distance_feature": {
          "field": "production_date",
          "pivot": "10d",
          "origin": "1514764800000",
          "boost": 3
        }
      }
    }
  }
}
```

I'm not clear on the distinction between `this.boost` and `boost`, but you can see them being multiplied in line 73 of `LongScriptFieldDistanceFeatureQuery`:

```
float weight = LongScriptFieldDistanceFeatureQuery.this.boost * boost;
```

### CODE REVIEW

File: docs/reference/query-dsl/distance-feature-query.asciidoc [hunk 1]
```diff
@@ -222,4 +222,4 @@ document's field value.
 Unlike the <<query-dsl-function-score-query,`function_score`>> query or other
 ways to change <<relevance-scores,relevance scores>>, the
 `distance_feature` query efficiently skips non-competitive hits when the
-<<search-uri-request,`track_total_hits`>> parameter is **not** `true`.
\ No newline at end of file
+<<search-uri-request,`track_total_hits`>> parameter is **not** `true`.
```
Comment (jrodewig) [inline docs/reference/query-dsl/distance-feature-query.asciidoc:225]: Not really related to this change, but we should probably put a newline here.
```suggestion
<<search-uri-request,`track_total_hits`>> parameter is **not** `true`.

```

**Review by jrodewig — COMMENTED** [review]
Thanks for catching this @aetter. I left some suggestions, but this LGTM overall.

Before merging, I'd like to get a review from @nik9000, who updated this query with #60851.

@nik9000 can you confirm the following:
- `weight` is `boost^2`
- This update affects only `master` and `7.x`

**Comment by aetter** [comment]
Thanks a bunch @jrodewig. Committed your suggestions. It isn't clear to me if weight really should be boost^2 or if the correct fix is to leave the docs as-is and tweak the calculation in code, but both seem like good outcomes.

**Comment by nik9000** [comment]
The code itself spits out:
```
        return Explanation.match(score, "Distance score, computed as weight * pivotDistance / (pivotDistance + abs(value - origin)) from:",
            Explanation.match(boost, "weight"),
            Explanation.match(pivotDistance, "pivotDistance"),
            Explanation.match(origin, "origin"),
            Explanation.match(value, "current value"));
```

I think, maybe, weight being `boost * boost` is a bug. We wrap the built query in a `BoostingQuery` and, deeper down, so does Lucene.

**Comment by nik9000** [comment]
I've filed and issue about the boost * boost thing and hopefully someone more knowledgeable than I am will reply.

**Comment by jrodewig** [comment]
Thanks @nik9000. I'm going to convert this to a draft until we determine if `boost^2` is desired or not. That'll prevent me or others from accidentally merging. Thanks again for raising this @aetter.

**Comment by mayya-sharipova** [comment]
@aetter  Thanks so much for finding out about this issue. 
@jrodewig  @nik9000 This is indeed a bug, and boost should be applied only once. We will fix it.  So we should not update the documentation.

**Comment by jrodewig** [comment]
Thanks @mayya-sharipova. I'll go ahead and close this PR out. Folks who are interested in the bug can follow #63691.

**Pull request closed without merging**